includes/admin/class-amp-editor-blocks.php

```diff
@@ -204,4 +212,110 @@ public function print_dirty_amp_scripts() {
 			wp_scripts()->do_items( $this->content_required_amp_scripts );
 		}
 	}
+
+	/**
+	 * Registers the dynamic block Latest Stories.
+	 * Much of this is taken from the Core block Latest Posts.
+	 *
+	 * @see register_block_core_latest_posts()
+	 */
+	public function register_block_latest_stories() {
+		register_block_type(
+			'amp/amp-latest-stories',
+			array(
+				'attributes'      => array(
+					'className'     => array(
+						'type' => 'string',
+					),
+					'storiesToShow' => array(
+						'type'    => 'number',
+						'default' => 5,
+					),
+					'order'         => array(
+						'type'    => 'string',
+						'default' => 'desc',
+					),
+					'orderBy'       => array(
+						'type'    => 'string',
+						'default' => 'date',
+					),
+				),
+				'render_callback' => array( $this, 'render_block_latest_stories' ),
+			)
+		);
+	}
+
+	/**
+	 * Renders the dynamic block Latest Stories.
+	 * Much of this is taken from the Core block Latest Posts.
+	 *
+	 * @see render_block_core_latest_posts()
+	 * @param array $attributes The block attributes.
+	 * @return string $markup The rendered block markup.
+	 */
+	public function render_block_latest_stories( $attributes ) {
+		$args = array(
+			'post_type'        => AMP_Story_Post_Type::POST_TYPE_SLUG,
+			'posts_per_page'   => $attributes['storiesToShow'],
+			'post_status'      => 'publish',
+			'order'            => $attributes['order'],
+			'orderby'          => $attributes['orderBy'],
+			'suppress_filters' => false,
+		);
+
+		$story_query = new WP_Query( $args );
+
+		ob_start();
+		foreach ( $story_query->posts as $post ) :
+			?>
+			<a href="<?php echo esc_url( get_permalink( $post ) ); ?>">
+				<?php echo get_the_post_thumbnail( $post->ID, self::LATEST_STORIES_IMAGE_SIZE ); ?>
```

Comment:
What if there is no featured image? Should a fallback be used or should such posts be skipped altogether?

Reply:
Good questions. For now, 5e2c053f0eef3185900ed1c62dac0d69b295a165 only outputs the `<a>` with the image is `has_post_thumbnail()`. I'll think about a possible fallback.

Reply:
246333c7e08fbdc3b7e33f9844dc64ac50633252 uses a fallback of any image attached to a post, in case there's no featured image. A more thorough solution might involve parsing the `post_content` for images, like from the image block.